pyseir/inference/infer_rt.py

```diff
@@ -0,0 +1,460 @@
+        for previous_day, current_day in zip(timeseries.index[:-1], timeseries.index[1:]):
+            # (5a) Calculate the new prior
+            current_prior = process_matrix @ posteriors[previous_day]
+
+            # (5b) Calculate the numerator of Bayes' Rule: P(k|R_t)P(R_t)
+            numerator = likelihoods[current_day] * current_prior
+
+            # (5c) Calcluate the denominator of Bayes' Rule P(k)
+            denominator = np.sum(numerator)
+
+            # Execute full Bayes' Rule
+            posteriors[current_day] = numerator / denominator
+
+            # Add to the running sum of log likelihoods
+            log_likelihood += np.log(denominator)
+
+        self.log_likelihood = log_likelihood
+
+        if plot:
+            plt.figure(figsize=(12, 8))
+            plt.plot(posteriors, alpha=0.1, color='k')
+            plt.grid(alpha=0.4)
+            plt.xlabel('$R_t$', fontsize=16)
+            plt.title('Posteriors', fontsize=18)
+        start_idx = -len(posteriors.columns)
+        return dates[start_idx:], times[start_idx:], posteriors
+
+    def infer_all(self, plot=False, shift_deaths=0):
+        """
+        Infer R_t from all available data sources.
+
+        Parameters
+        ----------
+        plot: bool
+            If True, generate a plot of the inference.
+        shift_deaths: int
+            Shift the death time series by this amount with respect to cases
+            (when plotting only, does not shift the returned result).
+
+        Returns
+        -------
+        inference_results: pd.DataFrame
+            Columns containing MAP estimates and confidence intervals.
+        """
+        df_all = None
+        available_timeseries = []
+
+        if len(self.get_timeseries(TimeseriesType.NEW_CASES)) > 0:
+            available_timeseries.append(TimeseriesType.NEW_CASES)
+
+        if len(self.get_timeseries(TimeseriesType.NEW_DEATHS)) > 0:
+            available_timeseries.append(TimeseriesType.NEW_DEATHS)
+
+        if self.hospitalization_data_type is load_data.HospitalizationDataType.CURRENT_HOSPITALIZATIONS:
+            # We have converted this timeseries to new hospitalizations.
+            available_timeseries.append(TimeseriesType.NEW_HOSPITALIZATIONS)
+        elif self.hospitalization_data_type is load_data.HospitalizationDataType.CUMULATIVE_HOSPITALIZATIONS:
+            available_timeseries.append(TimeseriesType.NEW_HOSPITALIZATIONS)
+
+        for timeseries_type in available_timeseries:
+
+            df = pd.DataFrame()
+            dates, times, posteriors = self.get_posteriors(timeseries_type)
+            if posteriors is not None:
+                df[f'Rt_MAP__{timeseries_type.value}'] = posteriors.idxmax()
+                for ci in self.confidence_intervals:
+                    ci_low, ci_high = self.highest_density_interval(posteriors, ci=ci)
+                    df[f'Rt_ci{int(100 * (1 - ci))}__{timeseries_type.value}'] = ci_low
```

Comment:
Maybe just change to `Rt_ci{int(100 * ci)}_low`/`Rt_ci{int(100 * ci)}_high`? Lower bound of 95% interval should be Rt_ci2.5 instead of Rt_ci5 right?

Reply:
I changed to 90% CIs and accepted your changes above.